server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -1,65 +1,97 @@
 package sttp.tapir.server.netty.internal
 
+import com.typesafe.netty.http.{DefaultStreamedHttpResponse, StreamedHttpRequest}
 import com.typesafe.scalalogging.Logger
 import io.netty.buffer.{ByteBuf, Unpooled}
 import io.netty.channel._
 import io.netty.handler.codec.http._
 import io.netty.handler.stream.{ChunkedFile, ChunkedStream}
+import org.reactivestreams.Publisher
 import sttp.monad.MonadError
 import sttp.monad.syntax._
 import sttp.tapir.server.model.ServerResponse
-import sttp.tapir.server.netty.{NettyResponse, NettyServerRequest, Route}
 import sttp.tapir.server.netty.NettyResponseContent.{
   ByteBufNettyResponseContent,
   ChunkedFileNettyResponseContent,
   ChunkedStreamNettyResponseContent
 }
+import sttp.tapir.server.netty.{NettyResponse, NettyResponseContent, NettyServerRequest, Route}
 
 import scala.collection.JavaConverters._
 
 class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit]) => Unit)(implicit me: MonadError[F])
-    extends SimpleChannelInboundHandler[FullHttpRequest] {
+    extends SimpleChannelInboundHandler[HttpRequest] {
 
   private val logger = Logger[NettyServerHandler[F]]
 
-  override def channelRead0(ctx: ChannelHandlerContext, request: FullHttpRequest): Unit = {
+  override def channelRead0(ctx: ChannelHandlerContext, request: HttpRequest): Unit = {
     if (HttpUtil.is100ContinueExpected(request)) {
       ctx.writeAndFlush(new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.CONTINUE))
       ()
     } else {
-      val req = request.retain()
+      request match {
+        case full: FullHttpRequest =>
+          val req = full.retain()
```

Comment:
is the only difference between those branches that we call retain/release in one case, and not in the other? If so, maybe we can extract the common part to a method